src/backend/distributed/utils/resource_lock.c

```diff
@@ -1060,3 +1086,218 @@ CitusLockTableAclCheck(Oid relationId, LOCKMODE lockmode, Oid userId)
 
 	return aclResult;
 }
+
+
+/*
+ * AcquireDistributedLockOnRelations_Internal acquire a distributed lock on worker nodes
+ * for given list of relations ids. Worker node list is sorted so that the lock
+ * is acquired in the same order regardless of which node it was run on.
+ * Notice that no lock is acquired on coordinator node if the coordinator is not
+ * added to the metadata.
+ *
+ * Notice that no validation is done on the relationIds, that is the responsibility of
+ * AcquireDistributedLockOnRelations.
+ *
+ * A nowait flag is used to require the locks to be available immediately
+ * and if that is not the case, an error will be thrown
+ */
+static void
+AcquireDistributedLockOnRelations_Internal(List *relationIdList, LOCKMODE lockMode, bool
+										   nowait)
+{
+	Oid relationId = InvalidOid;
+	List *workerNodeList = ActivePrimaryNodeList(NoLock);
+	Oid userId = GetUserId();
+
+	const char *lockModeCString = LockModeToLockModeCString(lockMode);
+
+	/*
+	 * We want to acquire locks in the same order across the nodes.
+	 */
+	workerNodeList = SortList(workerNodeList, CompareWorkerNodes);
+
+	UseCoordinatedTransaction();
+
+	int32 localGroupId = GetLocalGroupId();
+
+	foreach_oid(relationId, relationIdList)
```

Comment:
does the order of the locks matter? I mean, we calculate these ids, but could it be possible that on a concurrent `the same` operation from another locks the tables in a different order. Is that possible? 

I think sorting is also tricky. We wouldn't want the order that the user typed. Maybe it is best to keep as-is.

Reply:
> Can you elaborate on how sorting is tricky?

Again, if we send LOCK command to the nodes, we would avoid the sorting related edge cases. We'd be sending the command in a single round-trip, the same as how user typed. In that case, we'd be in-par with PG (given we already acquire commands in order across the workers as well)